include/drivers/hwinfo.h

```diff
@@ -30,6 +30,18 @@
 extern "C" {
 #endif
 
+enum reset_cause {
+	RESET_PIN				= BIT(0),
+	RESET_SOFTWARE				= BIT(1),
+	RESET_BROWNOUT				= BIT(2),
+	RESET_POWER				= BIT(3),
+	RESET_WATCHDOG				= BIT(4),
+	RESET_DEBUG				= BIT(5),
+	RESET_SECURITY				= BIT(6),
+	RESET_LOW_POWER_WAKE			= BIT(7),
+	RESET_CPU_LOCKUP			= BIT(8)
```

Comment:
Any reason to have both `RESET_WATCHDOG` and `RESET_CPU_LOCKUP`? In general, "CPU lockup" would almost always mean the same thing as "reset by watchdog."

Also there should be something like `RESET_ECC_FAIL` since this is a quite common feature found in the (automotive and industrial) safety MCUs.

Reply:
I came up with this list by looking through the docs for most/all the SoCs that are supported by Zephyr. CPU Lockup is different than Watchdog. It is asserted by ARM cores when there is an error while executing NMI or HardFault handlers. 

I will add ECC/Parity flags if you can point me to an example SoC.

Reply:
> I came up with this list by looking through the docs for most/all the SoCs that are supported by Zephyr and selecting the best common denominator. CPU Lockup is different than Watchdog. It is asserted by ARM cores when there is an error while executing NMI or HardFault handlers.

So, I suppose a more generic and technically correct name would be "CPU fault" then. AFAICS, the term "CPU lockup" in this context is only used by the ARM docs and a handful of SoCs.

> I will add ECC/Parity flags if you can point me to an example SoC.

* Atmel SAM D/E 5x series (e.g. ATSAME54)
* Atmel SAM E/V 7x series (e.g. ATSAME70)
* TI Hercules family (e.g. TMS570, RM)

Reply:
I kept the name same as the docs, but making it generic is not a bad idea.

> Atmel SAM D/E 5x series (e.g. ATSAME54)
Atmel SAM E/V 7x series (e.g. ATSAME70)
TI Hercules family (e.g. TMS570, RM)

👍